scripts/create-tenant.sh

```diff
@@ -0,0 +1,231 @@
+
+Your organisation is \"${ORG}\" and your login and password:
+
+ - login: ${EMAIL}
+ - password: ${PASSWORD}
+
+To begin using the PaaS you will need to install the cf command line client
+and execute the following commands.
+
+cf target https://api.cloud.service.gov.uk
+cf login
+
+Further documentation on the cf command line client is available from
+http://docs.cloudfoundry.org/cf-cli/install-go-cli.html
+
+
+Regards,
+Government PaaS team.
+'
+
+
+###########################################################################
+usage() {
+  cat <<EOF
+Usage:
+
+  $SCRIPT [-r] -e <email> -o <orgname>
+
+$SCRIPT will create a user and organisation in the CF service where you
+are currently logged in and send an email to the user if the password changes.
+
+Nothing will change if the organisation or the user already exists
+(unless the -r flag is used, which recreates the user with a new password).
+This way you can add a user to multiple organisations by running the script
+multiple times.
+
+Requirements:
+
+ * You must login with the cf client with an "admin" user.
+ * You must have a functional aws client with credentials configured.
+
+Where:
+  -r           Delete/recreate the user. The user will be recreated
+               and the password reset.
+
+  -e <email>   User email to add and configure as organization and
+               space manager. If the user is created or recreated,
+               they will receive a mail with the new password.
+
+  -o <orgname> Organisation to create and add the user to. If the
+               organisation already exists the script will carry on.
+
+EOF
+  exit 1
+}
+
+load_colors() {
+  if [ -t 1 ] ; then
+    export ESC_SEQ="\x1b["
+    export COL_RESET="${ESC_SEQ}39;49;00m"
+    export COL_LIGHT="${ESC_SEQ}1m"
+    export COL_DIM="${ESC_SEQ}2m"
+    export COL_BLINK="${ESC_SEQ}5m"
+    export COL_RED="${ESC_SEQ}31m"
+    export COL_GREEN="${ESC_SEQ}32m"
+    export COL_YELLOW="${ESC_SEQ}33m"
+    export COL_BLUE="${ESC_SEQ}34m"
+    export COL_MAGENTA="${ESC_SEQ}35m"
+    export COL_CYAN="${ESC_SEQ}36m"
+  fi
+}
+
+abort() {
+  echo -e "${COL_RED:-}${COL_LIGHT:-}ERROR:${COL_RESET:-} $*" 1>&2
+  exit 1
+}
+
+abort_usage() {
+  echo -e "${COL_RED:-}${COL_LIGHT:-}ERROR:${COL_RESET:-} $*" 1>&2
+  usage
+  exit 1
+}
+
+
+check_params_and_environment() {
+  if [ -z "${EMAIL:-}" ]; then
+    abort_usage "Email must be defined"
+  fi
+
+  local email_expr="^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}$"
+  if ! [[ "${EMAIL}" =~ ${email_expr} ]]; then
+    abort "You must specify a valid email"
+  fi
+
+  if [ -z "${ORG:-}" ]; then
+    abort_usage "Org must be defined"
+  fi
+
+  if ! aws ses get-send-quota >/dev/null 2>&1; then
+    abort "You must have AWS cli installed and configured with valid credentials. Test it with: aws ses get-send-quota"
+  fi
+
+}
+
+generate_password() {
+  PASSWORD=$(tr -dc _A-Z-a-z-0-9 < /dev/urandom | head -c32 || true)
```

Comment:
Fixed. I added the `LC_CTYPE=C` variable to avoid tr get confused.

I did not use openssl to avoid another dependency.